src/Sarif.Converters/ContrastSecurityConverter.cs

```diff
@@ -762,10 +788,18 @@ private string BuildSourcesString(HashSet<Tuple<string, string>> sources)
             var sb = new StringBuilder();
             foreach (Tuple<string, string> tuple in sources)
             {
-                if (tuple.Item1 == null || tuple.Item2 == null) { continue; }
+                if (sb.Length > 0) { sb.Append(", "); }
+
                 // Item1 is the name, Item2 is the source type, e.g., parameter
-                sb.Append(tuple.Item1 + "(" + tuple.Item2 + ")");
+                if (!string.IsNullOrWhiteSpace(tuple.Item1)) { sb.Append(tuple.Item1 + ": "); }
+
+                string sourceType = !string.IsNullOrWhiteSpace(tuple.Item2)
+                    ? tuple.Item2
+                    : "<unknown source type>";
```

Comment:
>"<unknown source type>" [](start = 22, length = 23)

you created a method scoped const string below for the System prefix but not here. is there a method you use to determine when it's appropriate to do this?

Reply:
No, I was just lazy here, will fix.

---
In reply to: [282150892](https://github.com/microsoft/sarif-sdk/pull/1480#discussion_r282150892) [](ancestors = 282150892)

Reply:
Fixed.

---
In reply to: [282152025](https://github.com/microsoft/sarif-sdk/pull/1480#discussion_r282152025) [](ancestors = 282152025,282150892)